### PR TITLE
[RISCV] Add processor definition for XiangShan-KunMingHu

### PR DESCRIPTION
**Pull request by Khao7342**
This pull request adds definitions for the XiangShan-KunMingHu processor. "XiangShan" is a high-performance open-source RISC-V processor project, and "KunMingHu" architecture is its third generation. Official documentation can be found at: [documentation](https://xiangshan-doc.readthedocs.io/zh-cn/latest/).

Currently, the KunMingHu core supports"RV64IMAFDCV_zba_zbb_zbc_zbs_zbkb_zbkc_zbkx_zknd_zkne_zknh_zksed_zksh_svinval_zicbom_zicboz_zicsr_zifencei". The scheduler model and other components will be submitted in subsequent patches.

### CODE REVIEW

**Comment by wangpc-pp** [comment]
Has KunMingHu's RTl been finalized (IIRC, we have developing vector unit)? And can we have different doc for different generations of XiangShan?

**Comment by camel-cdr** [comment]
Has the target architecture been finalized? (As in what it should be, not necessarily the rtl)

Just yesterday, there was a significant change in vector execution units:

[before](https://github.com/OpenXiangShan/XiangShan/blob/e25e4d90505c592524b410b127fe611ac49a3adf/src/main/scala/xiangshan/Parameters.scala#L355):
```
VFEX0: VfaluCfg, VfmaCfg, VialuCfg, VimacCfg
VFEX1: VipuCfg, VppuCfg, VfcvtCfg, F2vCfg, F2fCfg, F2iCfg, VSetRvfWvfCfg
VFEX2: VfaluCfg, VfmaCfg, VialuCfg
VFEX3: VfdivCfg, VidivCfg
```
[after](https://github.com/OpenXiangShan/XiangShan/blob/2e61107/src/main/scala/xiangshan/Parameters.scala#L357):
```
VFEX0: VfmaCfg, VialuCfg, VimacCfg, VppuCfg
VFEX1: VfaluCfg, VfcvtCfg, VipuCfg, VSetRvfWvfCfg
VFEX2: VfmaCfg, VialuCfg, F2vCfg
VFEX3: VfaluCfg, VfcvtCfg
VFEX4: VfdivCfg, VidivCfg
VFEX5: VfdivCfg, VidivCfg
```

**Comment by Khao7342** [comment]
> Has KunMingHu's RTL been finalized (IIRC, we have vector unit under development)? And can we have different doc for different generations of XiangShan?

Thanks for your attention to Xiangshan. KunMingHu's RTL has not been finalized yet. The development work on vectors is in the final stages of testing and performance optimization. And there won't be too many changes on the RTL.

**Comment by Khao7342** [comment]
> Has the target architecture been finalized? (As in what it should be, not necessarily the rtl)
> 
> Just yesterday, there was a significant change in vector execution units:
> 
> [before](https://github.com/OpenXiangShan/XiangShan/blob/e25e4d90505c592524b410b127fe611ac49a3adf/src/main/scala/xiangshan/Parameters.scala#L355):
> 
> ```
> VFEX0: VfaluCfg, VfmaCfg, VialuCfg, VimacCfg
> VFEX1: VipuCfg, VppuCfg, VfcvtCfg, F2vCfg, F2fCfg, F2iCfg, VSetRvfWvfCfg
> VFEX2: VfaluCfg, VfmaCfg, VialuCfg
> VFEX3: VfdivCfg, VidivCfg
> ```
> 
> [after](https://github.com/OpenXiangShan/XiangShan/blob/2e61107/src/main/scala/xiangshan/Parameters.scala#L357):
> 
> ```
> VFEX0: VfmaCfg, VialuCfg, VimacCfg, VppuCfg
> VFEX1: VfaluCfg, VfcvtCfg, VipuCfg, VSetRvfWvfCfg
> VFEX2: VfmaCfg, VialuCfg, F2vCfg
> VFEX3: VfaluCfg, VfcvtCfg
> VFEX4: VfdivCfg, VidivCfg
> VFEX5: VfdivCfg, VidivCfg
> ```

Thanks for your attention to Xiangshan. KunMingHu's RTL has not been finalized yet. The development work on vectors is in the final stages of testing and performance optimization. And there won't be too many changes on the RTL.

File: llvm/lib/Target/RISCV/RISCVProcessors.td [hunk 1]
```diff
@@ -378,3 +378,30 @@ def XIANGSHAN_NANHU : RISCVProcessorModel<"xiangshan-nanhu",
...
+                                               FeatureStdExtZbs,
+                                               FeatureStdExtZkn,
+                                               FeatureStdExtZksed,
+                                               FeatureStdExtZksh,
+                                               FeatureStdExtSvinval,
+                                               FeatureStdExtZicbom,
+                                               FeatureStdExtZicboz,
+                                               FeatureStdExtV,
+                                               FeatureStdExtZvl128b],
+                                               [TuneNoDefaultUnroll,
```
Comment (dtcxzyw) [inline llvm/lib/Target/RISCV/RISCVProcessors.td:404]: Can you provide some performance data about these options?
IIRC KunMingHu core supports the `lui + addi` fusion.

**Pull request closed without merging**